src/error.rs

```diff
@@ -0,0 +1,41 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum CrabError {
```

Comment:
I think there should be a separate error type for target and one for symbol.

Reply:
Does there still need to be a top-level wrapper for `pub type CrabResult<T> = Result<T, ErrorType>;`?

Reply:
Or should each mod (target and symbol) have their own `Result` type?

Reply:
> Does there still need to be a top-level wrapper for pub type CrabResult<T> = Result<T, ErrorType>;? Something like

That could be useful.